R/Data-methods.R

```diff
@@ -30,48 +33,73 @@ setMethod(
   definition = function(x, y, blind = FALSE, legend = TRUE, ...) {
     assert_flag(blind)
     assert_flag(legend)
+    h_plot_data_dataordinal(x, blind, legend, ...)
+  }
+)
 
-    if (x@nObs == 0L) {
-      return()
-    }
-
-    df <- h_plot_data_df(x, blind, ...)
-
-    p <- ggplot(df, aes(x = patient, y = dose)) +
-      geom_point(aes(shape = toxicity, colour = toxicity), size = 3) +
-      scale_colour_manual(
-        name = "Toxicity", values = c(Yes = "red", No = "black")
-      ) +
-      scale_shape_manual(name = "Toxicity", values = c(Yes = 17, No = 16)) +
-      scale_x_continuous(breaks = df$patient, minor_breaks = NULL) +
-      scale_y_continuous(
-        breaks = sort(unique(c(0, df$dose))),
-        minor_breaks = NULL,
-        limits = c(0, max(df$dose) * 1.1)
-      ) +
-      xlab("Patient") +
-      ylab("Dose Level")
-
-    p <- p + h_plot_data_cohort_lines(df$cohort, placebo = x@placebo)
-
-    if (!blind) {
-      p <- p +
-        geom_text(
-          aes(label = ID, size = 2),
-          data = df,
-          hjust = 0,
-          vjust = 0.5,
-          angle = 90,
-          colour = "black",
-          show.legend = FALSE
-        )
+#' Plot Method for the [`DataOrdinal`] Class
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' A method that creates a plot for [`DataOrdinal`] object.
+#'
+#' @param x (`DataOrdinal`)\cr object we want to plot.
+#' @param y (`missing`)\cr missing object, for compatibility with the generic
+#'   function.
+#' @param blind (`flag`)\cr indicates whether to blind the data.
+#'   If `TRUE`, then placebo subjects are reported at the same level
+#'   as the active dose level in the corresponding cohort,
+#'   and DLTs are always assigned to the first subjects in a cohort.
+#' @param legend (`flag`)\cr whether the legend should be added.
+#' @param tox_labels (`named list of character`)\cr The labels of the toxicity
+#' categories
+#' @param tox_shapes (`names list of integers`)\cr The symbols used to identify
+#' the toxicity categories
+#' @param ... not used.
+#'
+#' @note With more than 9 toxicity categories, toxicity symbols must be
+#' specified manually.\cr With more than 5 toxicity categories, toxicity labels
+#' must be specified manually.
```

Comment:
```suggestion
#' @note With more than 9 toxicity categories, toxicity symbols must be
#' specified manually. With more than 5 toxicity categories, toxicity labels
#' must be specified manually.
```
Could we align this, such that it is the same threshold (not one 9 and one 5 but both 9)? Then it would be easier to read/maintain maybe

Reply:
Done.  reduced to five because I doubt using more than 5 toxicity categories would make much sense.